modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -762,12 +762,41 @@ public ObjectEntry partialUpdateObjectEntry(
 		}
 
 		if (objectEntry.getProperties() != null) {
-			Map<String, Object> properties =
+			Map<String, Object> existingProperties =
 				existingObjectEntry.getProperties();
 
-			properties.putAll(objectEntry.getProperties());
+			Map<String, Object> properties = objectEntry.getProperties();
 
-			existingObjectEntry.setProperties(() -> properties);
+			for (ObjectRelationship objectRelationship :
+					_objectRelationshipLocalService.
+						getObjectRelationshipsByObjectDefinitionId2(
+							objectDefinition.getObjectDefinitionId())) {
+
+				ObjectField objectField =
+					objectFieldLocalService.getObjectField(
+						objectRelationship.getObjectFieldId2());
```

Comment:
@p-albuquerque to improve this, I would create a new finder in the service.xml:

```
<finder name="ODI_BT" return-type="Collection">
	<finder-column name="objectDefinitionId" />
	<finder-column name="businessType" />
</finder>
```

and a new method in the local service:

```
@Override
public List<ObjectField> getObjectFieldsByBusinessType(
	long objectDefinitionId, String businessType) {

	return _getObjectFields(
		objectFieldPersistence.findByODI_BT(
			objectDefinitionId, businessType));
}
```

to allow us to do:

```
for (ObjectField objectField :
		objectFieldLocalService.getObjectFieldsByBusinessType(
			objectDefinition.getObjectDefinitionId(),
			ObjectFieldConstants.BUSINESS_TYPE_RELATIONSHIP)) {

	String objectRelationshipERCObjectFieldName =
		ObjectFieldSettingUtil.getValue(
			ObjectFieldSettingConstants.
				NAME_OBJECT_RELATIONSHIP_ERC_OBJECT_FIELD_NAME,
			objectField);

	if (properties.containsKey(objectField.getName()) &&
		!properties.containsKey(
			objectRelationshipERCObjectFieldName)) {

		existingProperties.remove(
			objectRelationshipERCObjectFieldName);
	}
}
```